zhmcclient/_session.py

```diff
@@ -85,11 +90,11 @@ def __init__(self, host, userid=None, password=None):
             host=self._host,
             port=_HMC_PORT)
         self._headers = _STD_HEADERS  # dict with standard HTTP headers
-        self._session_id = None  # HMC session ID
+        self._session_id = session_id  # HMC session ID
+        if self._session_id is not None:
+            self._headers['X-API-Session'] = self._session_id
         self._session = None  # requests.Session() object
```

Comment:
The `Session._do_logon()` method sets the `_session` instance variable. I think we should also do this if a session ID was supplied. After all, we want to behave the same way as if a logon had been performed.